exchange/utils.go

```diff
@@ -980,9 +1020,11 @@ func applyFPD(fpd map[openrtb_ext.BidderName]*firstpartydata.ResolvedFirstPartyD
 				//BuyerUID needs to be set back to fpd before applying this fpd to final bidder request
 				fpdToApply.User.BuyerUID = reqWrapper.User.BuyerUID
 			}
-			// copy reqWrapper.User.EIDs to fpdToApply.User.EIDs
-			// because EIDs might have been removed by removeUnpermissionedEids.
-			fpdToApply.User.EIDs = reqWrapper.User.EIDs
+
+			// if FPD config didn't have user.eids - use reqWrapper.User.EIDs after removeUnpermissionedEids
+			if userEIDsOverride {
```

Comment:
As mentioned in an earlier comment, I suggest changing this variable name to `fpdUserEIDsExist` so the meaning is clearer.

Also, isn't this conditional supposed to be negated: `if !fpdUserEIDsExist`?

Reply:
Addressed